src/plugins/orangehrmCorporateDirectoryPlugin/Api/EmployeeDirectoryAPI.php

```diff
@@ -0,0 +1,243 @@
+            RequestParams::PARAM_TYPE_QUERY,
+            self::FILTER_MODEL,
+            self::MODEL_DEFAULT
+        );
+        return self::MODEL_MAP[$model];
+    }
+
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_EMP_NUMBER,
+                new Rule(Rules::IN_ACCESSIBLE_EMP_NUMBERS)
+            ),
+            $this->getModelParamRule(),
+        );
+    }
+
+    /**
+     * @return ParamRule
+     */
+    protected function getModelParamRule(): ParamRule
+    {
+        return $this->getValidationDecorator()->notRequiredParamRule(
+            new ParamRule(
+                self::FILTER_MODEL,
+                new Rule(Rules::IN, [array_keys(self::MODEL_MAP)])
+            )
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointCollectionResult
+    {
+        $employeeDirectoryParamHolder = new EmployeeDirectorySearchFilterParams();
+        $this->setSortingAndPaginationParams($employeeDirectoryParamHolder);
+
+        $empNumber = $this->getRequestParams()->getIntOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::FILTER_EMP_NUMBER
+        );
+        if (!is_null($empNumber)) {
+            $employeeDirectoryParamHolder->setEmpNumbers([$empNumber]);
+        } else {
+            $accessibleEmpNumbers = $this->getUserRoleManager()->getAccessibleEntityIds(Employee::class);
+            $employeeDirectoryParamHolder->setEmpNumbers($accessibleEmpNumbers);
+        }
+
+        $employeeDirectoryParamHolder->setJobTitleId(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_JOB_TITLE_ID
+            )
+        );
+        $employeeDirectoryParamHolder->setLocationId(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_LOCATION_ID
+            )
+        );
+
+        $employees = $this->getEmployeeDirectoryService()->getEmployeeDirectoryDao()->getEmployeeList(
+            $employeeDirectoryParamHolder
+        );
+        $count = $this->getEmployeeDirectoryService()->getEmployeeDirectoryDao()->getEmployeeCount(
+            $employeeDirectoryParamHolder
+        );
+        return new EndpointCollectionResult(
+            $this->getModelClass(),
+            $employees,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::FILTER_EMP_NUMBER,
+                    new Rule(Rules::IN_ACCESSIBLE_EMP_NUMBERS)
```

Comment:
```suggestion
                    new Rule(Rules::ENTITY_ID_EXISTS, [Employee::class]),
```
may have to replace like this. since all users can access any employee in the system

Reply:
done